imap-codec/benches/greeting.rs

```diff
@@ -0,0 +1,70 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use imap_codec::{decode::Decoder, encode::Encoder, imap_types::response::Greeting, GreetingCodec};
+use imap_types::{
+    auth::AuthMechanism,
+    core::{Text, Vec1},
+    response::{Capability, Code, GreetingKind},
+};
+
+fn criterion_benchmark(c: &mut Criterion) {
+    // # Setup
+    let codec = GreetingCodec::new();
+    let instances = [("simple", create_simple()), ("complex", create_complex())];
+
+    for (instance, object) in instances {
+        c.bench_function(
+            format!("bench_greeting_serialize_{instance}").as_str(),
+            |b| {
+                b.iter(|| {
+                    serialize(&codec, &object);
+                })
+            },
+        );
+
+        let input = serialize(&codec, &object);
+        c.bench_function(format!("bench_greeting_parse_{instance}").as_str(), |b| {
+            b.iter(|| {
+                parse(&codec, black_box(&input[..]));
```

Comment:
Because of the `;` the return value will not be returned to criterion. I think this might prevent the expression from being evaluated.